client/components/payment-methods-list/payment-method.js

```diff
@@ -32,14 +32,21 @@ const PaymentMethod = ( {
 	onUncheckClick,
 	className,
 	isAllowingManualCapture,
+	required,
+	locked,
```

Comment:
Thanks for raising these concerns, Ismael. They are valid, and I agree we should try to keep our components' props on a short leash and prevent bloat.

I've thought more about it, and I think in our situation, the `required` and `locked` are valid since they allow us to keep external concerns out of the `PaymentMethod` component.

If we were to embed the logic and get rid of the props, we would have something like this `const required = 'card' === id && isUpeEnabled;`

This would lead us to couple the logic to the global UPE state and make our component less agnostic. I think your example is just at the border of acceptable since it only has stylistic ramifications while being required or locked (especially) has logical implications.

On the upside of adding the extra props, unit testing the behavior of each is much more clear-cut.

Please let me know you thoughts on this, Ismael. Much appreciated.

Reply:
I think `PaymentMethod` component is already highly tied to UPE, not reused anywhere, and contains too many props and external/internal logic for a presentational component. At this point adding more props will not do any harm 🙂 

My point here was about how easy it is with React (in this case) to overengineer components trying to abstract everything with the idea of reusing it later. But sometimes we don't need a reusable component, and it's easier to read and maintain if we design it accordingly.

Anyway, your solution looks good 👍

Reply:
I agree about the overoptimization/over-abstraction tendency, and it is definitely something to look out for.